### PR TITLE
Add an uninstall.sh

### LINKED ISSUE #237: Lacking uninstaller
I run the installer.sh before recognize there's fonts-powerline in Ubuntu, but `installer.sh` doesn't provide uninstall functionality.

### PR DESCRIPTION
**Pull request by ialokim**
Up to now, there's an `install.sh` script that does the installation process, but it lacks uninstall functionality. I've created a little `uninstall.sh` to fill this gap.

Fixes #237

### CODE REVIEW

File: uninstall.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  font_dir="$HOME/Library/Fonts"
+else
+  # Linux
+  font_dir="$HOME/.local/share/fonts"
+  mkdir -p $font_dir
+fi
+
+# Remove all fonts to user fonts directory
+echo "Removing fonts..."
+eval $find_command | xargs -n1 -0 -I % sh -c 'test="%"; rm -f "'"$font_dir"'/${test##*/}"'
```
Comment (ZyX-I) [inline uninstall.sh]: I do not see why you should use `eval $find_command` here, but I can see why you should not: `"$var"` is not a proper escaping. Also don’t put directory names inside an argument via `%`, `sh -c` does accept arguments and this variant of passing arguments has no problems with escaping. Also don’t use `"$font_dir"` there as well, export variable instead.

File: uninstall.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# Reset font cache on Linux
+if command -v fc-cache @>/dev/null ; then
```
Comment (ZyX-I) [inline uninstall.sh]: Why not `which` and what `@>` is supposed to mean? `which fc-cache >/dev/null 2>&1` normally works fine and if you spell that this way (and replace `[[ … ]]` with `test` above) then script will be free of bashism.

File: uninstall.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# Set source and target directories
+powerline_fonts_dir=$( cd "$( dirname "$0" )" && pwd )
```
Comment (ZyX-I) [inline uninstall.sh]: Need quotes to be consistent. I know that they are not actually needed in variable assignments, but better have them then not.

File: uninstall.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+# if an argument is given it is used to select which fonts to install
```
Comment (ZyX-I) [inline uninstall.sh]: Comment talks about installing.

File: uninstall.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+prefix="$1"
+
+find_command="find \"$powerline_fonts_dir\" \( -name '$prefix*.[o,t]tf' -or -name '$prefix*.pcf.gz' \) -type f -print0 "
+
+if [[ `uname` == 'Darwin' ]]; then
```
Comment (ZyX-I) [inline uninstall.sh]: Use `$()` to stay consistent with `$()` above.

File: uninstall.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  # MacOS
+  font_dir="$HOME/Library/Fonts"
+else
+  # Linux
+  font_dir="$HOME/.local/share/fonts"
+  mkdir -p $font_dir
```
Comment (ZyX-I) [inline uninstall.sh]: You are doing uninstallation, if font directory does not exists *exit* with a message, not create it.

File: uninstall.sh [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+fi
+
+# Remove all fonts to user fonts directory
+echo "Removing fonts..."
+eval $find_command | xargs -n1 -0 -I % sh -c 'test="%"; rm -f "'"$font_dir"'/${test##*/}"'
+
+# Reset font cache on Linux
+if command -v fc-cache @>/dev/null ; then
+    echo "Resetting font cache, this may take a moment..."
+    fc-cache -f $font_dir
```
Comment (ZyX-I) [inline uninstall.sh]: Missing quotes.

File: README.rst [hunk 8]
```diff
@@ -25,6 +25,12 @@ You can copy and paste the commands to your terminal. Comments are fine too.
     cd ..
     rm -rf fonts
 
+Uninstall
+---------
+
+Run ``./uninstall.sh`` to uninstall all Powerline Fonts. You can also copy
```
Comment (ZyX-I) [inline README.rst]: Missing the fact that `./uninstall.sh` needs to be run with the same HEAD in the repository as fonts were installed, fonts are sometimes being removed from repository or renamed.

File: uninstall.sh (hunk 2 above)
Comment (ialokim) [reply uninstall.sh]: I'm okay with this.

File: uninstall.sh (hunk 5 above)
Comment (ialokim) [reply uninstall.sh]: Done.

File: uninstall.sh (hunk 4 above)
Comment (ialokim) [reply uninstall.sh]: Fixed.

File: uninstall.sh (hunk 6 above)
Comment (ialokim) [reply uninstall.sh]: You're completely right, fixed that!

File: uninstall.sh (hunk 3 above)
Comment (ialokim) [reply uninstall.sh]: Added.

File: uninstall.sh (hunk 7 above)
Comment (ialokim) [reply uninstall.sh]: Fixed.

File: README.rst (hunk 8 above)
Comment (ialokim) [reply README.rst]: Fixed.

File: uninstall.sh (hunk 1 above)
Comment (ialokim) [reply uninstall.sh]: The `eval` is necessary to support the `or` switch in the command string. But I've exported the variables as pointed out.

**Comment by ialokim** [comment]
Thanks for your review, I've fixed your issues, while three of them were also present in the current `install.sh`. I've added a second commit fixing them, too.

**Comment by ZyX-I** [comment]
> The eval is necessary to support the or switch in the command string. But I've exported the variables as pointed out.

How it is? It should work fine with double-quoted `$prefix` if you move replace eval with actual command. BTW, what does comma do in `.[o,t]tf`?

**Comment by ialokim** [comment]
> How it is? It should work fine with double-quoted $prefix if you replace eval with actual command. BTW, what does comma do in .[o,t]tf?

As already stated, I've copied essentially the `install.sh` and changed only the necessary lines to provide uninstall functionality. I've now replaced the `eval` with the actual command as you've suggested and changed the `install.sh` accordingly, too.

File: install.sh [hunk 9]
```diff
@@ -1,14 +1,12 @@
...
-powerline_fonts_dir=$( cd "$( dirname "$0" )" && pwd )
+powerline_fonts_dir="$( cd "$( dirname "$0" )" && pwd )"
 
 # if an argument is given it is used to select which fonts to install
 prefix="$1"
 
-find_command="find \"$powerline_fonts_dir\" \( -name '$prefix*.[o,t]tf' -or -name '$prefix*.pcf.gz' \) -type f -print0"
-
-if [[ `uname` == 'Darwin' ]]; then
+if test $(uname) == 'Darwin' ; then
```
Comment (ZyX-I) [inline install.sh]: `test` uses single equals sign. And `$()` needs to be escaped, or there is some small potential to get “missing argument after Darwin” error (`$(uname)` will expand into nothing if `uname` is missing).

File: uninstall.sh [hunk 10]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Set source and target directories
+powerline_fonts_dir="$( cd "$( dirname "$0" )" && pwd )"
+
+# if an argument is given it is used to select which fonts to uninstall
+prefix="$1"
+
+if test $(uname) == 'Darwin' ; then
```
Comment (ZyX-I) [inline uninstall.sh]: Same here.

File: uninstall.sh [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  # MacOS
+  font_dir="$HOME/Library/Fonts"
+else
+  # Linux
+  font_dir="$HOME/.local/share/fonts"
+  if test ! -d "$font_dir"; then
```
Comment (ZyX-I) [inline uninstall.sh]: Above there is space before semicolon.

File: install.sh (hunk 9 above)
Comment (ialokim) [reply install.sh]: Fixed.

File: uninstall.sh (hunk 11 above)
Comment (ialokim) [reply uninstall.sh]: Added the space here, too.

**Pull request merged**